Actions/CheckForUpdates/yamlclass.ps1

```diff
@@ -185,4 +224,255 @@ class Yaml {
+        $steps = $this.GetStepsFromJob($job)
+        if (!$steps) {
+            Write-Host "::Warning::Cannot find job '$job'"
+            return
+        }
+        $anchorIdx = $steps.IndexOf($anchorStep)
+        if ($anchorIdx -lt 0) {
+            Write-Host "::Warning::Cannot find anchor step '$anchorStep' in job '$job'"
+            return
+        }
+        foreach($customStep in $customSteps | Where-Object { $_.AnchorStep -eq $anchorStep -and $_.Before -eq $before }) {
+            if ($steps -contains $customStep.Name) {
+                Write-Host "Custom step '$($customStep.Name)' already exists in job '$job'"
+            }
+            else {
+                $anchorStart = 0
+                $anchorCount = 0
+                if ($this.Find("Jobs:/$($job):/steps:/- name: $($anchorStep)", [ref] $anchorStart, [ref] $anchorCount)) {
+                    if ($before) {
+                        $this.Insert($anchorStart-1, @('') + @($customStep.Content | ForEach-Object { "      $_" }))
+                    }
+                    else {
+                        $this.Insert($anchorStart+$anchorCount, @('') + @($customStep.Content | ForEach-Object { "      $_" }))
+                    }
+                }
+            }
+            # Use added step as anchor for next step
+            $anchorStep = $customStep.Name
+        }
+    }
+
+    [void] AddCustomStepsToYaml([string] $job, [hashtable[]] $customSteps, [hashtable[]] $anchors) {
+        foreach($anchor in $anchors) {
+            $this.AddCustomStepsToAnchor($job, $customSteps, $anchor.Step, $anchor.Before)
+        }
+    }
+
+    static [PSCustomObject] GetPermissionsFromArray([string[]] $permissionsArray) {
+        $permissions = [PSCustomObject]@{}
+        $permissionsArray | ForEach-Object {
+            $permissions | Add-Member -MemberType NoteProperty -Name $_.Split(':')[0].Trim() -Value $_.Split(':')[1].Trim()
+        }
+        return $permissions
+    }
+
+    static [string[]] GetPermissionsArray([PSCustomObject] $permissions) {
+        $permissionsArray = @()
+        $permissions.PSObject.Properties.Name | ForEach-Object {
+            $permissionsArray += "$($_): $($permissions."$_")"
+        }
+        return $permissionsArray
+    }
+
+    static [PSCustomObject] MergePermissions([PSCustomObject] $permissions, [PSCustomObject] $permissions2) {
+        $permissions2.PSObject.Properties.Name | ForEach-Object {
+            if ($permissions.PSObject.Properties.Name -eq $_) {
+                $permission = $permissions."$_"
+                $permission2 = $permissions2."$_"
+                if ($permission -eq 'write' -or $permission2 -eq 'write') {
+                    $permissions."$_" = 'write'
+                }
+                elseif ($permission -eq 'read' -or $permission2 -eq 'read') {
+                    $permissions."$_" = 'read'
+                }
+                else {
+                    $permissions."$_" = 'none'

```

Comment:
Would we set the permission to None if one workflow has `actions: read` and the other has `actions: write`? 🤔

Reply:
No, if either we (or a customization) has write permissions - we set write.
Then, if either we (or a customization has read permissions - we set read.
Else we set none.